modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1562,6 +1562,19 @@ void GDScriptAnalyzer::resolve_assignable(GDScriptParser::AssignableNode *p_assi
 				parser->push_warning(p_assignable->initializer, GDScriptWarning::NARROWING_CONVERSION);
 #endif
 			}
+
+			if (p_assignable->initializer->is_constant && type_from_metatype(initializer_type).builtin_type != type_from_metatype(specified_type).builtin_type) {
```

Comment:
I'm not sure about this condition, especially metatypes (but I haven't found any errors with enums in manual testing).

Reply:
> but why `type_from_metatype` is needed?

Without this, I get the error:

```gdscript
# Cannot convert variable "a" initializer value of type int to the specified type test.gd::MyEnum.
var a: MyEnum = 1
```

Reply:
I see, yes, that's the problem with `resolve_datatype`, needs to be fixed (by other PR).

But I don't think you need to call `type_from_metatype` on `initializer_type` since it is an actual type.